store/tikv/coprocessor.go

```diff
@@ -977,6 +995,28 @@ func (worker *copIteratorWorker) handleCopResponse(bo *Backoffer, rpcCtx *RPCCon
 			}
 		}
 	}
+	if resp.pbResp.IsCacheHit && cacheValue != nil {
+		// Cache hit and is valid: use cached data as response data and we don't update the cache.
+		data := make([]byte, len(cacheValue.Data))
+		copy(data, cacheValue.Data)
+		resp.pbResp.Data = data
+	} else {
+		// Cache not hit or cache hit but not valid: update the cache if the response can be cached.
+		if cacheKey != nil && resp.pbResp.CacheLastVersion > 0 {
+			if worker.store.coprCache.CheckAdmission(resp.pbResp.Data.Size(), resp.detail.ProcessTime) {
+				data := make([]byte, len(resp.pbResp.Data))
+				copy(data, resp.pbResp.Data)
+
+				newCacheValue := coprCacheValue{
+					Data:              data,
+					TimeStamp:         worker.req.StartTs,
```

Comment:
maybe better use max start_ts of `data` if convenient, it's corner case maybe won't fix

```
T1:  -----------begin(ts=1) ------------- select v from t where id = 1 [fill cache with ts=1] -----
T2:  begin(ts=0) -------- sleep ------------------------ select v from t where id = 1[should reuse cached but refill?]
```

Reply:
Do you mean that for T2 it should not overwrite T1's entity in the cache?